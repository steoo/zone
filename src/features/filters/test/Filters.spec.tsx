import React from 'react';
import renderWithStore from '../../../app/utils/testing';
import Filters from '../Filters.component';
import { initialState } from '../../movies/Movies.slice';
import { movies } from '../../movies/test/Movies.fixtures';
import { genres } from '../../genres/test/Genres.fixtures';

describe('Components', () => {
  describe('Features', () => {
    describe('Filters', () => {
      const mockState = {
        movies: {
          ...initialState,
          data: movies
        },
        genres: {
          isFetching: false,
          data: genres
        }
      };

      it('should render, without crashing', () => {
        const { baseElement } = renderWithStore(<Filters genres={genres} />, {
          initialState: mockState
        });

        expect(baseElement).toMatchSnapshot();
      });

      it('should have current vote average set to 4', () => {
        const localMockState = {
          ...mockState,
          movies: {
            ...mockState.movies,
            filters: {
              ...mockState.movies.filters,
              avg: 4
            }
          }
        };

        const { baseElement, getByLabelText } = renderWithStore(<Filters genres={genres} />, {
          initialState: localMockState
        });

        expect(getByLabelText('number')).toHaveValue(4);
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
