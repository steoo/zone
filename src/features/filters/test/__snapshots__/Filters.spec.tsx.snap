// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Features Filters should have current vote average set to 4 1`] = `
<body>
  <div>
    <div
      class="sc-dlfnbm edICMC"
    >
      <div
        class="sc-hKgILt kZKIbv"
      >
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Action
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Adventure
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Animation
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Comedy
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Crime
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Documentary
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Drama
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Family
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Fantasy
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          History
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Horror
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Music
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Mystery
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Romance
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Science Fiction
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          TV Movie
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Thriller
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          War
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Western
        </button>
      </div>
      <div
        class="sc-eCssSg dOfNQx"
      >
        <h3>
          Vote average: 
        </h3>
        <input
          aria-label="number"
          type="number"
          value="4"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Components Features Filters should render, without crashing 1`] = `
<body>
  <div>
    <div
      class="sc-dlfnbm edICMC"
    >
      <div
        class="sc-hKgILt kZKIbv"
      >
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Action
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Adventure
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Animation
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Comedy
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Crime
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Documentary
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Drama
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Family
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Fantasy
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          History
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Horror
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Music
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Mystery
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Romance
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Science Fiction
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          TV Movie
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Thriller
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          War
        </button>
        <button
          class="sc-gsTCUz dOAjnQ"
        >
          Western
        </button>
      </div>
      <div
        class="sc-eCssSg dOfNQx"
      >
        <h3>
          Vote average: 
        </h3>
        <input
          aria-label="number"
          type="number"
          value="0"
        />
      </div>
    </div>
  </div>
</body>
`;
